### PR TITLE
Adds unique validation to alternative identifiers

### PR DESCRIPTION
**Pull request by awead**
## Description

Valkyrie does not enforce any restrictions about alternate identifiers. It assumes that the downstream application will do whatever work is needed. To that end, we need to ensure that the alternate identifiers we provide are unique system-wide.

Connected to #824 

## Changes

* Creates a `Unique` validator and assigns it to the `alternate_ids` property in the data dictionary
* Changes the way validators are invoked from the change set so that the current change set is passed to the validate, as well as the field it is checking. This enables us to determine if the field is being updated from an existing value, or is a new, unsaved record.

### CODE REVIEW

File: spec/cho/validation/member_spec.rb [hunk 1]
```diff
@@ -3,19 +3,35 @@
...
     end
 
     context 'with a non-existent id' do
       subject { described_class.new('non-existent-id') }
 
-      its(:exists?) { is_expected.to be(false) }
+      its(:exists_by_id?) { is_expected.to be(false) }
+    end
+  end
+
```
Comment (cam156) [inline spec/cho/validation/member_spec.rb]: The validation of the exists? alias is untested Are you intending to keep that alias?

File: spec/cho/work/submissions/edit_spec.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@
 require 'rails_helper'
 
 RSpec.describe 'Editing works', type: :feature do
-  let(:resource) { create(:work, :with_creator, title: 'Work to edit', work_type_id: work_type.id) }
+  let(:resource) { create(:work, :with_creator, :with_metadata, title: 'Work to edit', work_type_id: work_type.id) }
```
Comment (cam156) [inline spec/cho/work/submissions/edit_spec.rb]: This seems like an odd change.  The setup changed, but nothing in the test changed.  It looks to me like you are now requiring the alternate id, but it is not required to save.

Comment (awead) [reply spec/cho/work/submissions/edit_spec.rb]: @cam156 the alt. id was empty previously. I needed to add it to verify we can update works with one. The `:with_metadata` trait is really a nudge that we ought to start filling in all the fields in our forms to get an idea of things will break or not. The update vs. create issue with unique alt. ids never would have surfaced if I hadn't put one on a resource.

File: spec/cho/validation/member_spec.rb (hunk 1 above)
Comment (awead) [reply spec/cho/validation/member_spec.rb]: I've added a test.

**Pull request merged**